s3/s3.go

```diff
@@ -832,8 +835,13 @@ func (o *Object) Hash(t fs.HashType) (string, error) {
 	etag := strings.Trim(strings.ToLower(o.etag), `"`)
 	// Check the etag is a valid md5sum
 	if !matchMd5.MatchString(etag) {
-		// fs.Debugf(o, "Invalid md5sum (probably multipart uploaded) - ignoring: %q", etag)
-		return "", nil
+		o.readMetaData()
+		if hash, ok := o.meta[metaMD5Hash]; ok {
+			hashBytes, _ := base64.StdEncoding.DecodeString(*hash)
```

Comment:
S3 expects `Content-MD5` to be expressed in base64, and also demonstrates expressing the user-defined `md5chksum` in base64.  I convert `md5chksum` to base64 here [[1](https://aws.amazon.com/premiumsupport/knowledge-center/data-integrity-s3/)].